common/nodejs-utils/src/application.rs

```diff
@@ -55,20 +42,48 @@ pub fn warn_prebuilt_modules(app_dir: &Path, _warn_later: &WarnGuard<Stdout>) {
         log_warning_later(formatdoc! {"
             Warning: {node_modules} checked into source control
 
-            Add these files and directories to {gitignore}. See the Dev Center for more info: 
+            Add these files and directories to {gitignore}. See the Dev Center for more info:
             https://devcenter.heroku.com/articles/node-best-practices#only-git-the-important-bits
         ", node_modules = fmt::value("node_modules"), gitignore = fmt::value(".gitignore") });
     }
 }
 
 #[derive(Debug)]
 pub enum Error {
+    MissingLockfile,
     MultipleLockfiles(Vec<PackageManager>),
 }
 
 impl Display for Error {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
+            Error::MissingLockfile => {
+                writedoc!(
+                    f,
+                    "
+                        Couldn't determine Node.js package manager. Package manager lockfile not found.
+
+                        A lockfile from a supported package manager is required to install Node.js
```

Comment:
I may be wrong but not ending this sentence with a trailing `\` will end up producing a `\n` here in the message being output which could end up being displayed funny in a terminal. If so, I'd suggest keeping your paragraphs intact by adding the trailing `\` characters.  E.g.;

```rust
"
    A lockfile from a supported package manager is required to install Node.js \
    dependencies. The package.json for this project specifies dependencies, but \
    there isn't a lockfile.
"
```

Reply:
Good call. Fixed in c823276.